source/server-discovery-and-monitoring/tests/rs/application_error_handling.yml

```diff
@@ -0,0 +1,351 @@
+description: Detect stale application errors
+uri: mongodb://a/?replicaSet=rs
+phases:
+- description: Primary A is discovered
+  responses:
+  - - a:27017
+    - ok: 1
+      ismaster: true
+      hosts:
+      - a:27017
+      setName: rs
+      minWireVersion: 0
+      maxWireVersion: 9
+      topologyVersion:
+        processId:
+          "$oid": '000000000000000000000001'
+        counter:
+          "$numberLong": '1'
+  outcome:
+    servers:
+      a:27017:
+        type: RSPrimary
+        setName: rs
+        topologyVersion:
+          processId:
+            "$oid": '000000000000000000000001'
+          counter:
+            "$numberLong": '1'
+        pool:
+          generation: 0
+    topologyType: ReplicaSetWithPrimary
+    logicalSessionTimeoutMinutes: null
+    setName: rs
+
+- description: Ignore stale application error (topologyVersion less)
+  applicationErrors:
+  - address: a:27017
+    generation: 0
+    error:
+      ok: 0
+      errmsg: ShutdownInProgress
+      code: 91
```

Comment:
Please add cases for all 7 of the recovering/non-primary error codes/messages, as well as one other error code for a negative test.

Reply:
I will add this once we agree on the format. (I want to avoid churn.)